x-pack/plugins/stack_alerts/server/alert_types/index_threshold/alert_type.ts

```diff
@@ -211,5 +216,19 @@ export function getAlertType(
       alertInstance.scheduleActions(ActionGroupId, actionContext);
       logger.debug(`scheduled actionGroup: ${JSON.stringify(actionContext)}`);
     }
+
+    options.services.getRecoveredAlertIds().forEach((recoveredAlertId: string) => {
+      const agg = params.aggField ? `${params.aggType}(${params.aggField})` : `${params.aggType}`;
+      const humanFn = `${agg} is NOT ${getHumanReadableComparator(
+        params.thresholdComparator
+      )} ${params.threshold.join(' and ')}`;
+      const baseContext: BaseActionContext = {
+        date,
+        group: recoveredAlertId,
```

Comment:
The hard part (in general) is going to be getting the new "value" which no longer meets the threshold.  Turns out, index threshold is kinda dumb because it doesn't bake the threshold test into the ES query, it requests all the values and evaluates the condition in JS.  We should probably change that, and let ES do the work.

But in this case, it's great!  Because we actually do have the new value that doesn't meet the threshold.  So we should add it in here.

I wonder how many rules will actually be able to provide the "current" values of items for the context.  I'm hesitant to tell rule authors to design rules like this one, since they are by their nature inefficient.

I guess ideally, you'd want to do one query, where you pass in the existing active instances, and the query runs like it used to, but also returns the "values" for items that were passed in as well.  Is that just two separate aggs in a single query?  Then you could run through the the `getRecoveredAlertIds()` and fill in the values from the new agg.

Reply:
Yeah I agree, this is not ideal. We don't want the executors to be doing extra queries to get information for the recovery alerts. And getting access to the previous value of the alert requires persisting the context as well as the state, which I don't think we want to be doing either.

I'm actually thinking the long term solution is we use the rule registry and alerts as data. The lifecycle executor is doing basically what we need for the recovery context, which is querying the AAD index for the alert, which gives you all the values written out during the previous execution that you can use in the recovery context. Not sure if this issue can wait for that stuff to get lifted in the framework, so I'm trying to see if we have any shorter term options.